### PR TITLE
fixes #110

### LINKED ISSUE #110: downloadFirmwareBinary and downloadProductFirmware don't check this.prefix
it appears both of these calls are probably broken, they use this.prefix (most other calls use this in request builder which checks to see if it's defined).  Bringing the check from build request to those calls to fix the download functions.

### PR DESCRIPTION
**Pull request by dmiddlecamp**
checks the prefix which can otherwise be undefined, and would prevent these functions from working as expected.

### CODE REVIEW

File: src/Particle.js [hunk 1]
```diff
@@ -590,7 +590,9 @@ class Particle {
 	downloadFirmwareBinary({ binaryId, auth }){
 		const uri = `/v1/binaries/${binaryId}`;
 		const req = request('get', uri);
-		req.use(this.prefix);
+		if (this.prefix) {
+			req.use(this.prefix);
+		}
 		this.headers(req, auth);
```
Comment (busticated) [inline src/Particle.js]: i was seeing failures related to this line as well - see [CH story](https://app.clubhouse.io/particle/story/45862/downloadproductfirmware-and-downloadfirmwarebinary-fail-w-typeerror-fn-is-not-a-function) - are you?

**Comment by dmiddlecamp** [comment]
Hmm, now getting errors around "this.headers", these functions might have more issues than expected, looking into it.

**Comment by dmiddlecamp** [comment]
Okay, got downloadFirmwareBinary working, and made compatible changes to 'downloadProductFirmware', but I haven't tested that other call yet.

**Comment by dmiddlecamp** [comment]
Steps to Test:

Here's the code I'm running for an internal testing tool, things you'll need:

* a test app
* an access token

```
//my test app
void setup() {
    Particle.publish("status", "Empty app, no product");
}

void loop() { }

```


```
var express = require('express');
var router = express.Router();

var path = require('path');
var fs = require('fs');
var when = require('when');
var pipeline = require('when/pipeline');
var settings = require('../settings.js');
var Hogan = require('hogan.js');
var Particle = require('particle-api-js');

var data = req.body;
  try {
    console.log('hi!  I got... ', req.body);

    // WARNING: Hey!  YOU!  Don't run anything like this in an untrusted user facing capacity
    // this is just for a testing tool!

    // NOTE: if you were going to make it user facing, don't like, let them specify any old file.
    var filename = data.file;
    var filename_only = path.basename(filename);


    // Get and compile our template
    //

    var contents = fs.readFileSync(filename).toString();
    var template = Hogan.compile(contents);


    // Render our template
    //

    var template_params = data.template;
    var rendered_contents = template.render(template_params);

    // Save it in the temp folder
    //

    var new_path = path.join(__dirname, '../public/javascripts/tests/iota/firmware_temp');
    var new_filename = path.join(new_path, filename_only);
    fs.writeFileSync(new_filename, rendered_contents);


    var compileParams = {
      files: {},
      platformId: data.platformId,
      auth: settings.particle_access_token
    };
    compileParams.files[filename_only] = new_filename;


    var particle = new Particle();


    pipeline([
        function() {
          return particle.compileCode(compileParams);
        },

        function(data) {
          console.log("success ? ", data);
          var binary_id = data.body.binary_id;
          return particle.downloadFirmwareBinary({ binaryId: binary_id, auth: settings.particle_access_token});
        },

        function(data) {
          console.log("binary downloaded?  Got back ", data);
          res.send({ok: true, msg: "done!"});
        }
    ]).catch(function(ex) {
      console.log("pipeline error ", ex);

      res.send({ok: false, error: ex });
    });
    
  }
  catch(ex) {
    console.log("exception thrown during compile request handling ", ex);
  }

```

**Comment by dmiddlecamp** [comment]
okay, I think I removed the errant file, hopefully I didn't mess up the branch too bad

**Comment by busticated** [comment]
@dmiddlecamp apologies for the churn here - i had to tweak some stuff to accommodate testing given the current tooling, etc. do you mind trying these changes quick before i merge?

**Pull request merged**